### PR TITLE
New Action: Update Info Plist (App Identifier and Display Name)

### PR DESCRIPTION
**Pull request by tobiasstrebitzer**
Features:
- The update_info_plist action allows changing the app/bundle identifier and display name. This is useful if you create different apps from a single lane, using the same source project.

```
+------------------------------------------------------------------+
|                        update_info_plist                         |
+------------------------------------------------------------------+
| Update a Info.plist file with bundle identifier and display name |
|                                                                  |
| Created by tobiasstrebitzer                                      |
+------------------------------------------------------------------+

+----------------+--------------------------------+--------------------------------+
|                                update_info_plist                                 |
+----------------+--------------------------------+--------------------------------+
| Key            | Description                    | Env Var                        |
+----------------+--------------------------------+--------------------------------+
| xcodeproj      | Path to your Xcode project     | FL_UPDATE_PLIST_PROJECT_PATH   |
| plist_path     | Path to info plist             | FL_UPDATE_PLIST_PATH           |
| app_identifier | The App Identifier of your app | FL_UPDATE_PLIST_APP_IDENTIFIER |
| display_name   | The Display Name of your app   | FL_UPDATE_PLIST_DISPLAY_NAME   |
+----------------+--------------------------------+--------------------------------+
```

### CODE REVIEW

File: lib/fastlane/actions/update_info_plist.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                
+        # Check if parameters are set
+        if params[:app_identifier] or params[:display_name]
+          
+          # Assign folder from parameter or search for xcodeproj file
+          folder = params[:xcodeproj] || Dir["*.xcodeproj"].first
+        
+          # Read existing plist file 
+          info_plist_path = File.join(folder, "..", params[:plist_path])
+          raise "Couldn't find info plist file at path '#{params[:plist_path]}'".red unless File.exists?(info_plist_path)
```
Comment (KrauseFx) [inline lib/fastlane/actions/update_info_plist.rb]: As far as I understand the user has to provide both a path to a project and a path to the plist. Why not do that in one parameter? 

If we leave it like this, we'd still have to change the output probably:

from
```ruby
raise "Couldn't find info plist file at path '#{params[:plist_path]}'".red unless File.exists?(info_plist_path)
```
to
```ruby
raise "Couldn't find info plist file at path '#{info_plist_path}'".red unless File.exists?(info_plist_path)
```

File: lib/fastlane/actions/update_info_plist.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          
+          # Update plist values
+          plist['CFBundleIdentifier'] = params[:app_identifier] if params[:app_identifier]
+          plist['CFBundleDisplayName'] = params[:display_name] if params[:display_name]
+          
+          # Write changes to file
+          plist_string = Plist::Emit.dump(plist)
+          File.write(info_plist_path, plist_string)
+          
+          Helper.log.info "Updated #{params[:plist_path]} 💾.".green
```
Comment (KrauseFx) [inline lib/fastlane/actions/update_info_plist.rb]: :floppy_disk:  :smile:

**Comment by KrauseFx** [comment]
Thank you @tobiasstrebitzer, I added a comment, almost ready to merge :+1:

**Comment by KrauseFx** [comment]
@tobiasstrebitzer could you update the pull request to address my above comment? I'm curios why the user needs to provide multiple paths.

**Comment by u16suzu** [comment]
It is nice pull request!
Great job :+1:

**Comment by KrauseFx** [comment]
Thanks again @tobiasstrebitzer for this great pull request :rocket: :+1:

**Comment by u16suzu** [comment]
Thank you so much.
I use this action. :+1:

**Pull request merged**